tools/dep_updaters/update-zlib.sh

```diff
@@ -1,23 +1,46 @@
 #!/bin/sh
 set -e
-# Shell script to update zlib in the source tree to a specific version
+# Shell script to update zlib in the source tree to the most recent version.
+# Zlib rarely creates tags or releases, so we use the latest commit on the main branch.
+# See: https://github.com/nodejs/node/pull/47417
 
 BASE_DIR=$(cd "$(dirname "$0")/../.." && pwd)
 DEPS_DIR="$BASE_DIR/deps"
 
-CURRENT_VERSION=$(grep "#define ZLIB_VERSION" "$DEPS_DIR/zlib/zlib.h" | sed -n "s/^.*VERSION \"\(.*\)\"/\1/p")
- 
-NEW_VERSION_ZLIB_H=$(curl -s "https://chromium.googlesource.com/chromium/src/+/refs/heads/main/third_party/zlib/zlib.h?format=TEXT" | base64 --decode)
+NEW_UPSTREAM_SHA1=$(git ls-remote "https://chromium.googlesource.com/chromium/src/third_party/zlib.git" HEAD | awk '{print $1}')
```

Comment:
```suggestion
```

Reply:
`NEW_UPSTREAM_SHA1` is also used on line 11

Reply:
It can be replaced with `git rev-parse --short=7 FETCH_HEAD` after the fetch command.